solution/index.js

```diff
@@ -0,0 +1,554 @@
+        headers: {
+            'Content-Type': 'application/json',           
+        }
+    });    
+    
+    document.removeEventListener('mousemove', onMouseMove); 
+    loaderDisplay(false);     
+
+    if(response != undefined && !response.ok){
+        let errorString = "Load Error! => " + response.status + " " + response.statusText;
+        displayError(true, errorString);
+    }
+    else{
+        //recieve tasks from server, update local storage
+        let tasksData = await response.json();             
+        localStorage.clear();
+    
+
+        localStorage.setItem("tasks",tasksData.tasks); 
+        displayError(true, "Tasks Loaded Succesfully","notification");
+    }
+}
+
+
+// ==================================
+// ======  Functions ============
+// ==================================
+
+function loaderDisplay(display){
+    // Creating:
+    //<div class="loader"><div></div><div></div><div></div><div></div></div>
+    if(display){
+        let body = document.querySelector("body");
+        let divArray = []; 
+        for (let i = 0; i < 4; i++) {
+            divArray.push(createElement("div"));   
+        }
+        let loaderDiv = createElement("div", divArray, ["loader"]);
+        const { clientX: mouseX, clientY: mouseY } = window.event;
+    
+        loaderDiv.style.top = `${mouseY}px`;
+        loaderDiv.style.left = `${mouseX}px`;        
+        body.append(loaderDiv);
+    }
+    else{
+        let loader = document.querySelector(".loader");
+        loader.remove();        
+    }    
+}
+
+async function LoadPageFromServer() {
+    await getServerTasks();
+    reloadTasksPage(searchInput.value);  
+} 
+
+function ResetLocalStorage() {
+    const tasks = {
+        "todo": [],
+        "in-progress": [],
+        "done": [],        
+    };
+    localStorage.setItem('tasks', JSON.stringify(tasks)); 
+}
+
+/* 
+*   ===> Adding task to LocalStorage <===
+*/
+function AddToSection(taskName, key, index = 0 ,showError = false){
+    const tasksObject = getLocalStorageTasks();    
+    const pSectionTasksArray = tasksObject[key];    
+    if(taskName === ""){
+        displayError(showError, "Add failed, task is empty.");    
+        return;
+    }
+    if(index === 0){
+        pSectionTasksArray.unshift(taskName);
+    }
+    else{
+        pSectionTasksArray.splice(index, 0, taskName);
+    }
+    //pass input validation tests
+    //update tasks and store in local storage        
+    localStorage.setItem('tasks', JSON.stringify(tasksObject)); 
+    displayError(showError,"Task was added succesfully","notification");    
+}
+
+// ===> gets id of input - return its value (as string) <===
+function getTextFromInputId(id){
+    let input = document.querySelector("#" + id);         
+    let inputValue = input.value;
+    // Erase input value
+    input.value = "";
+    return inputValue;
+}
+
+// refresh page according to local storage filtered by the query (search) text
+function reloadTasksPage(query){    
```

Comment:
There is no need to reboot all the entries once a change has been made.There are more convenient and efficient ways